apps/store/src/features/priceCalculator/PurchaseFormV2/OfferPresenterV2/ProductTierSelectorV2/ProductTierSelectorV2.tsx

```diff
@@ -33,91 +33,84 @@ export function ProductTierSelectorV2({ offers, selectedOffer, onValueChange }:
-              {offer.id === selectedOffer.id && (
-                <motion.div
-                  transition={{
-                    duration: framerTransitions.defaultDuration,
-                    ...framerTransitions.easeInOutCubic,
-                  }}
-                  initial={{ height: 0, marginTop: 0, opacity: 0 }}
-                  animate={{ height: 'fit-content', marginTop: theme.space.md, opacity: 1 }}
-                  exit={{ height: 0, marginTop: 0, opacity: 0 }}
-                >
-                  <Button
-                    variant="secondary"
-                    fullWidth={true}
-                    size="medium"
-                    onClick={openTierDialog}
+              <Text color="textSecondary" className={sprinkles({ marginTop: 'md' })}>
+                {getVariantDescription(offer.variant.typeOfContract)}
+              </Text>
+              <AnimatePresence initial={false}>
+                {offer.id === selectedOffer.id && (
+                  <motion.div
+                    transition={{
+                      duration: framerTransitions.defaultDuration,
+                      ...framerTransitions.easeInOutCubic,
+                    }}
+                    initial={{ height: 0, marginTop: 0, opacity: 0 }}
+                    animate={{ height: 'fit-content', marginTop: theme.space.md, opacity: 1 }}
+                    exit={{ height: 0, marginTop: 0, opacity: 0 }}
                   >
-                    {t('COMPARE_TIERS_LABEL')}
-                  </Button>
-                </motion.div>
-              )}
-            </AnimatePresence>
-          </CardRadioGroup.Item>
-        ))}
-      </CardRadioGroup.Root>
+                    <Dialog.Trigger asChild>
+                      <Button variant="secondary" fullWidth={true} size="medium">
+                        {t('COMPARE_TIERS_LABEL')}
+                      </Button>
+                    </Dialog.Trigger>
+                  </motion.div>
+                )}
+              </AnimatePresence>
+            </CardRadioGroup.Item>
+          ))}
+        </CardRadioGroup.Root>
+
+        <Dialog.Trigger asChild>
+          <Button
+            variant="secondary"
+            size="small"
+            className={sprinkles({ alignSelf: 'center' })}
+            Icon={<PlusIcon />}
+            iconPosition="right"
+          >
+            {t('COMPARE_COVERAGE_BUTTON')}
+          </Button>
+        </Dialog.Trigger>
 
-      <ComparisonTableModal
-        controlRef={tierDialogRef}
-        tiers={offers}
-        selectedTierId={selectedOffer.id}
-      >
-        <Button
-          variant="secondary"
-          size="small"
-          className={sprinkles({ alignSelf: 'center' })}
-          Icon={<PlusIcon />}
-          iconPosition="right"
-        >
-          {t('COMPARE_COVERAGE_BUTTON')}
-        </Button>
-      </ComparisonTableModal>
+        <TierComparisonDialog tiers={offers} selectedTierId={selectedOffer.id} />
```

Comment:
nit: maybe name it `TierComparisonDialogContent` since it's not all parts of a Dialog
```suggestion
        <TierComparisonDialogContent tiers={offers} selectedTierId={selectedOffer.id} />
```